x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.test.ts

```diff
@@ -330,7 +332,70 @@ describe('Host Isolation', () => {
   });
 
   describe('Cases', () => {
-    it.todo('logs a comment to the provided case');
-    it.todo('logs a comment to any cases associated with the given alert');
+    let casesClient: CasesClientMock;
+
+    beforeEach(async () => {
+      casesClient = (await endpointAppContextService.getCasesClient(
+        {} as KibanaRequest
+      )) as CasesClientMock;
+
+      let counter = 1;
+      casesClient.cases.getCasesByAlertID.mockImplementation(async () => {
+        return [
+          {
+            id: `case-${counter++}`,
+            title: 'case',
+          },
+        ];
+      });
+    });
+
+    it('logs a comment to the provided case', async () => {
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'], case_ids: ['one', 'two'] },
+      });
+
+      expect(casesClient.attachments.add).toHaveBeenCalledTimes(2);
+    });
+
+    it('logs a comment to any cases associated with the given alerts', async () => {
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'], alert_ids: ['one', 'two'] },
+      });
+
+      expect(casesClient.attachments.add).toHaveBeenNthCalledWith(
+        1,
```

Comment:
I don't think we actually care about order that the backend writes these comments in. Need to make this a little more permissive

Reply:
No, we don't care about the order, but do care that its called with the expected case ids.

Reply:
better then to collect up the arguments from all calls, and then check array membership